### ISSUE TITLE
t.trial.runner.DocTestCase is deprecated and should be removed

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5554]
|<img alt="rockstar's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| rockstar reported|
|-|-|
|Trac ID|trac#5554|
|Type|task|
|Created|2012-03-14 22:18:56Z|
|Branch|https://github.com/twisted/twisted/tree/remove-doctestcase-5554|

While fixing ticket [#5386](https://github.com/twisted/twisted/issues/5386), jesstess mentioned that the removal of DocTestCase should be its own ticket. This. This is that ticket.

Attachments:

* [5554.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/33d/33d6357cfaaf0f72991b0ecd8c56da066613c089/b40b37cd4c9364062f5704276b355e2ae1f96b7d.patch) (1238 bytes) - added by moijes12 on 2012-06-14 05:15:53Z - 
* [5554.2.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/33d/33d6357cfaaf0f72991b0ecd8c56da066613c089/7a0f08b6900c7b9681739e5f3d62391cddadc154.patch) (1272 bytes) - added by moijes12 on 2012-06-19 07:24:23Z -  Patch updated with the corrections to the .removal file.

<details><summary>Searchable metadata</summary>

```
trac-id__5554 5554
type__task task
reporter__rockstar rockstar
priority__normal normal
milestone__ 
branch__branches_remove_doctestcase_5554 branches/remove-doctestcase-5554
branch_author__moijes12 moijes12
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1331763536000000 1331763536000000
changetime__1342327755000000 1342327755000000
version__None None
owner__spiv spiv
cc__moijes12 cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="bartekci's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_1"></a>|bartekci set owner to bartekci<br>bartekci set status to `assigned`|
|-|-|

This was removed while working on [#5387](https://github.com/twisted/twisted/issues/5387). In Changeset [33807]

**Comment by twisted-trac** [comment]
|[<img alt="moijes12's avatar" src="https://avatars.githubusercontent.com/u/4566851?s=50" width="50" height="50">](https://github.com/moijes12)<a name="note_2"></a>|@moijes12 removed owner<br>@moijes12 set status to `new`|
|-|-|

I've attached the patch. I don't know about the below entries though I have a feeling they shouldn't matter or they aren't related.


twisted/trial/runner.py:580:            # Work around Python issue2604: DocTestCase.tearDown clobbers globs 

twisted/trial/unittest.py:1498:    methods. That is L{pyunit.FunctionTestCase} and L{pyunit.DocTestCase}. 

twisted/trial/unittest.py:1538:_docTestCase = getattr(doctest, 'DocTestCase', None)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_3"></a>|@thijstriemstra set owner to @moijes12|
|-|-|

Thanks. Can you create a branch for this patch and run it on the buildslaves using force-builds.py? The .removal file should also mention when it was deprecated (Twisted 8.0).

**Comment by twisted-trac** [comment]
|[<img alt="moijes12's avatar" src="https://avatars.githubusercontent.com/u/4566851?s=50" width="50" height="50">](https://github.com/moijes12)<a name="note_4"></a>|@moijes12 commented|
|-|-|

Hi 


I don't know how to create a branch and I don't really think I'll be able to do it as I don't have commit rights ? But I'll do the other change to the .removal file and upload it.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_5"></a>|@thijstriemstra commented|
|-|-|

Oh my bad, I thought you had commit rights. Someone else can apply the patch and/or put it in a branch in that case. Thanks.

**Comment by twisted-trac** [comment]
|[<img alt="moijes12's avatar" src="https://avatars.githubusercontent.com/u/4566851?s=50" width="50" height="50">](https://github.com/moijes12)|@moijes12 removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_7"></a>|@thijstriemstra commented|
|-|-|

(In [34644]) apply 5554.2.patch, refs [#5554](https://github.com/twisted/twisted/issues/5554)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_8"></a>|@thijstriemstra commented|
|-|-|

[Build results](http://buildbot.twistedmatrix.com/builders/fedora11-64bit-py2.7/builds/765).

**Comment by twisted-trac** [comment]
|<img alt="spiv's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_9"></a>|spiv set owner to spiv<br>spiv set status to `assigned`|
|-|-|

KILL IT KILL IT KILL IT WITH FIRE.

Er, what I meant to say: looks good to me.  I'll submit it.

**Comment by twisted-trac** [comment]
|<img alt="spiv's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_10"></a>|spiv set status to `closed`|
|-|-|

(In [34802]) Merge remove-doctestcase-5554: Remove twisted.trial.runner.DocTestCase.

Author: moijes12
Reviewer: thijs, spiv
Fixes: [#5554](https://github.com/twisted/twisted/issues/5554)

Removes twisted.trial.runner.DocTestCase, which has been deprecated since
Twisted 8.0.
